syntax = "proto3";

package user_certificate;

import "google/protobuf/timestamp.proto";

service UserCertificateService {
  rpc Create (CreateUserCertificateRequest)
      returns (UserCertificateResponse);

  rpc ListByUser (ListUserCertificatesRequest)
      returns (ListUserCertificatesResponse);

  rpc Delete (DeleteUserCertificateRequest)
      returns (DeleteUserCertificateResponse);
}

message CreateUserCertificateRequest {
  string user_id = 1;

  string title = 2;
  string issuer = 3;
  google.protobuf.Timestamp issued_at = 4;
  string link = 5;
}

message UserCertificateResponse {
  string id = 1;
  string user_id = 2;

  string title = 3;
  string issuer = 4;
  google.protobuf.Timestamp issued_at = 5;
  string link = 6;

  google.protobuf.Timestamp created_at = 7;
}

message ListUserCertificatesRequest {
  string user_id = 1;
  int32 limit = 2;
  int32 offset = 3;
}

message ListUserCertificatesResponse {
  repeated UserCertificateResponse items = 1;
  int64 total = 2;
}

message DeleteUserCertificateRequest {
  string certificate_id = 1;
}

message DeleteUserCertificateResponse {}
